ambry-cloud/src/test/java/com/github/ambry/cloud/azure/AzureContainerCompactorIntegrationTest.java

```diff
@@ -0,0 +1,148 @@
+  private AzureCloudConfig azureConfig;
+  private VerifiableProperties verifiableProperties;
+  private AzureCloudDestination cloudDestination;
+
+  @Before
+  public void setup() throws ReflectiveOperationException {
+    // TODO Create the required cosmos table as well as the required azure blob.
+    Properties testProperties = new Properties();
+    try (InputStream input = this.getClass().getClassLoader().getResourceAsStream(PROPS_FILE_NAME)) {
+      if (input == null) {
+        throw new IllegalStateException("Could not find resource: " + PROPS_FILE_NAME);
+      }
+      testProperties.load(input);
+    } catch (IOException ex) {
+      throw new IllegalStateException("Could not load properties from resource: " + PROPS_FILE_NAME);
+    }
+    testProperties.setProperty("clustermap.cluster.name", "Integration-Test");
+    testProperties.setProperty("clustermap.datacenter.name", "uswest");
+    testProperties.setProperty("clustermap.host.name", "localhost");
+    testProperties.setProperty("kms.default.container.key",
+        "B374A26A71490437AA024E4FADD5B497FDFF1A8EA6FF12F6FB65AF2720B59CCF");
+
+    testProperties.setProperty(CloudConfig.CLOUD_DELETED_BLOB_RETENTION_DAYS, String.valueOf(1));
+    testProperties.setProperty(AzureCloudConfig.AZURE_PURGE_BATCH_SIZE, "10");
+    verifiableProperties = new VerifiableProperties(testProperties);
+    CloudConfig cloudConfig = new CloudConfig(verifiableProperties);
+    this.azureConfig = new AzureCloudConfig(verifiableProperties);
+    MetricRegistry registry = new MetricRegistry();
+    CloudDestinationFactory cloudDestinationFactory =
+        Utils.getObj(cloudConfig.cloudDestinationFactoryClass, verifiableProperties, registry);
+    cloudDestination = (AzureCloudDestination) cloudDestinationFactory.getCloudDestination();
+  }
+
+  @After
+  public void destroy() throws IOException {
+    // TODO destroy the abs blob and cosmos db
+    if (cloudDestination != null) {
+      cloudDestination.close();
+    }
+  }
+
+  @Test
+  public void testUpdateDeletedContainer() throws CloudStorageException {
+    Set<Container> containers = generateContainers(5);
+    cloudDestination.updateDeletedContainers(containers, null);
+    verifyCosmosData(containers);
+    verifyCheckpoint(containers);
+  }
+
+  private void verifyCosmosData(Set<Container> containers) {
+    String query = "SELECT VALUE COUNT(1) FROM C";
+    SqlQuerySpec querySpec = new SqlQuerySpec(query);
+    Timer timer = new Timer();
+    cloudDestination.getCosmosDataAccessor()
+        .executeCosmosQuery(azureConfig.cosmosDeletedContainerCollectionLink, null, querySpec, new FeedOptions(),
+            timer);
+  }
+
+  private void verifyCheckpoint(Set<Container> containers) {
+
+  }
+
+  /**
+   * Generate specified number of {@link Container}s.
+   * @param numContainers number of {@link Container}s to generate.
+   * @return {@link Set} of {@link Container}s.
+   */
+  private Set<Container> generateContainers(int numContainers) {
```

Comment:
Please move the logic to generate a random container to AccountTestUtils.
This code looks copied from AccountContainerTest.initializeRefContainers.

Reply:
will do.. this class is work in progress right now.

Reply:
this is fixed now.